### DISCUSSION TITLE
Logging

### DISCUSSION THREAD

**Discussion by nexus667** [RakambdaOrg/ChannelPointsMiner#621] [Q&A]
How do you setup/enable logging to a log file in the config.json?

**Comment by Rakambda** [comment]
Logs are not setup withing the config.

* You'll have to pass a new parameter when starting the app, refering to a log configuration file : https://rakambdaorg.github.io/ChannelPointsMiner/miner/main/installation/clone.html
* Create the log configuration file : https://rakambdaorg.github.io/ChannelPointsMiner/miner/main/configuration/logger.html
  * Examples, including logging to a file : https://github.com/RakambdaOrg/ChannelPointsMiner/tree/develop/src/examples/loggers

**Reply by Ravendwyr** [reply]
Is there a way to customise or filter out certain messages from the output?

e.g. let's say I only want updates about Drop progression in the console; is there a way to filter out channel point gains and other messages?

**Reply by Rakambda** [reply]
> Is there a way to customise or filter out certain messages from the output?
> 
> 
> 
> e.g. let's say I only want updates about Drop progression in the console; is there a way to filter out channel point gains and other messages?

The logger can be configured to only allow certain classes to log things at a certain level.
I'm on a trip and it's hard to give examples, but let's say you have the logger named fr.rakambda.channelpointsminer.events.ClaimPoints that logs something, then you can set this logger to INFO, and the rest (root logger) to ERROR for example. This way you'll have error messages and the event you want (if several can set several loggers to info).
There's some examples in the source code (/src/example/resources if I remember well). It doesn't show this particular use case but there's several loggers set up, you just have to change their names according to what you wanna do.
